docs/proposal/2022-12-23-rbac.md

```diff
@@ -0,0 +1,399 @@
+
+      config:
+        defaultNamespace: default
+        restrictAccess: false
+
+  'kubectl-pods-rw':
+    botkube/kubectl:
+      enabled: true
+      context:
+        kubeconfig: &kubeconfigCtx # no need to specify it again if it's the same as above
+  'kubectl-deploy-rw':
+    botkube/kubectl@v1.0.0:
+      enabled: true
+      context:
+        kubeconfig: &kubeconfigCtx # no need to specify it again if it's the same as above
+```
+
+A few remarks:
+
+- Merging different mapping configuration won't not supported and will result in an error. The configuration will be validated during Botkube startup.
+- As Kubernetes impersonation requires an username when using group impersonation, the default user name for mapping will be set as a static `default` value, as shown in Option 4 above.
+- If the kubeconfig context is not provided (as a result, `mapping.type` is empty), Botkube won't create a temporary Kubeconfig for a given plugin.
+
+Initially, we'd support only `Static`, `UserEmail` and `ChannelName` mapping types. `UserGroupName` can be added later based on user demand.
+
+### Executing commands with proper access rights
+
+To ensure compatibility with our plugin system, we'll use [K8s API user impersonation](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation) to create a special short-living Kubeconfig to use for a given execution. This will be handled globally by Botkube for all plugins.
+
+We'll extend our gPRC API for executors in the following way:
+
+```proto
+message ExecuteRequest {
+  // Commands represents the exact command that was specified by the user.
+  string command = 1;
+  // Configs is a list of Executor configurations specified by users.
+  repeated Config configs = 2;
+
+  // New field
+  ExecuteContext context = 3;
+}
+
+message ExecuteContext {
+  string kubeconfig_path = 1;
+}
+```
+
+This path may be used by the plugin to access Kubernetes cluster in a restricted way.
+
+### Executor plugin isolation
+
+To ensure no plugin has access to the sensitive configuration data, like Slack or Discord tokens, we will modify the plugin manager to run command with in an isolated environment in the following way.
+
+For each executor plugin, We will use chroot to limit the access to the filesystem. A given plugin will have an isolated directory with all dependencies (e.g. `helm` or `kubectl` binary). For each command execution Botkube plugin manager will:
+
+- create a temporary subdirectory inside the isolated directory,
+- create a temporary Kubeconfig file in the subdirectory,
+- pass the Kubeconfig path via gRPC API,
+- delete the temporary subdirectory after command execution.
+
+This is possible as we have full control over how the command is run:
+
+```go
+	for key, path := range bins {
+    // ...
+
+		//nolint:gosec // warns us about 'Subprocess launching with variable', but we are the one that created that variable.
+		cmd := exec.Command(path)
```

Comment:
I would also set the `KUBECONFIG={path}` automatically.

Reply:
While I did that in PoC, we cannot do it here, as executor is a long-running subprocess, and we want to have a specific kubeconfig per a given execution. So the only option I see is the one described in the proposal - passing path as a part of the gRPC API. Even if we had some `envs` in the context, plugin couldn't set them they might process multiple commands at the same time and there could be a problem with using proper access rights.